File: 2015/3/3-1.py
with open('input.txt') as f:
  path = f.readline().strip('\n')

(x, y) = [0, 0]
visits = {}

for c in path:
  visits[(x, y)] = True
  if c == '>':
    x += 1
  elif c == '<':
    x -= 1
  elif c == '^':
    y += 1
  elif c == 'v':
    y -= 1
  visits[(x, y)] = True

print('Houses Visited: %d' % (len(visits.keys())))

File: 2015/3/3-1.go
package main

import (
    "fmt"
)

type visit struct {
    x int
    y int
}

func main() {

    script := getFileLines("input.txt")[0]

    visits := []visit{}

    addVisit(&visits, 0, 0)

    x := 0
    y := 0

    for _, c := range script {
        switch c {
        case '<':
            x -= 1
        case '>':
            x += 1
        case '^':
            y += 1
        case 'v':
            y -= 1
        default:
            fmt.Printf("Not sure about %v in script\n", c)
        }
        addVisit(&visits, x, y)
    }
    fmt.Printf("Visits: %v\n", len(visits))
    fmt.Printf("Unique Visits: %v\n", len(uniqueHouses(visits)))

}


func addVisit(visits *[]visit, x int, y int) {
    *visits = append(*visits, visit{x: x, y: y})
}

func uniqueHouses(visits []visit) ([]visit) {
    uniq := []visit{}
    for _, c := range visits {
        already := false
        for _, e := range uniq {
            if c.x == e.x && c.y == e.y {
                already = true
            }
        }
        if already == false {
            uniq = append(uniq, c)
        }
    }
    return uniq
}
